rl4co/utils/decoding.py

```diff
@@ -280,9 +284,8 @@ def pre_decoder_hook(
 
             td.set("action", action)
             td = env.step(td)["next"]
-            logprobs = torch.zeros_like(
-                td["action_mask"], device=td.device
-            )  # first logprobs is 0, so p = logprobs.exp() = 1
+            # first logprobs is 0, so p = logprobs.exp() = 1
+            logprobs = torch.zeros_like(action, device=td.device)
```

Comment:
Note: If we are storing all log p, I guess we should put either
` logprobs = torch.zeros_like(td["action_mask"])`, otherwise, it will be  `torch.zeros_like(action)`, right?

Reply:
Done

Reply:
great catch!